java-api-examples/README.md

```diff
@@ -0,0 +1,89 @@
+
+0.Introduction
+---
+Java wrapper `com.k2fsa.sherpaonnx.OnlineRecognizer` for `sherpa-onnx`. Java is a cross-platform language; you can build jni .so lib according to your system, and then use the same java api for all your platform.
+``` xml
+Depend on:
+  Openjdk 1.8
+```
+---
+1.Compile libsherpa-onnx-jni.so
+---
+Compile sherpa-onnx/jni/jni.cc according to your system.
+Example for Ubuntu 18.04 LTS, Openjdk 1.8.0_362:
+``` xml
+  git clone https://github.com/k2-fsa/sherpa-onnx
+  cd sherpa-onnx
+  mkdir build
+  cd build
+  cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DSHERPA_ONNX_ENABLE_JNI=ON ..
+  make -j6
+```
+---
+2.Download asr model files
+---
+[click here for more detail](https://k2-fsa.github.io/sherpa/onnx/pretrained_models/index.html)
+---
+3.Config model config.cfg
+---
+``` xml
+  #set libsherpa-onnx-jni.so lib root dir
+  solibpath=/sherpa-onnx/build/lib/libsherpa-onnx-jni.so  
+  #model config  
+  sample_rate=16000                  
+  feature_dim=80
+  rule1_min_trailing_silence=2.4
+  rule2_min_trailing_silence=1.2
+  rule3_min_utterance_length=20
+  encoder=/sherpa-onnx/build/bin/sherpa-onnx-streaming-zipformer-bilingual-zh-en-2023-02-20/encoder-epoch-99-avg-1.onnx
+  decoder=/sherpa-onnx/build/bin/sherpa-onnx-streaming-zipformer-bilingual-zh-en-2023-02-20/decoder-epoch-99-avg-1.onnx
+  joiner=/sherpa-onnx/build/bin/sherpa-onnx-streaming-zipformer-bilingual-zh-en-2023-02-20/joiner-epoch-99-avg-1.onnx
+  tokens=/sherpa-onnx/build/bin/sherpa-onnx-streaming-zipformer-bilingual-zh-en-2023-02-20/tokens.txt
+  num_threads=4
+  enable_endpoint_detection=false
+  decoding_method=greedy_search
+  max_active_paths=4
+```
+---
+4.A simple java example
+---
+refer to [java_api_example](https://github.com/zhaomingwork/sherpa-onnx/blob/java-wrapper-support/java-api-examples/src/DecodeFile.java) for more detail.
+``` java
+import com.k2fsa.sherpaonnx.OnlineRecognizer;
+import com.k2fsa.sherpaonnx.OnlineStream;
+String cfgpath=appdir+"/modelconfig.cfg";
+OnlineRecognizer.setCfgPath(cfgpath);      //set model config file
+OnlineRecognizer rcgOjb = new OnlineRecognizer();   //create a recognizer
+CreateStream streamObj=rcgOjb.CreateStream();       //create a stream for reading wav data
+float[] buffer = rcgOjb.readWavFile(wavfilename); // read data from file
+streamObj.acceptWaveform(buffer, 16000);          //feed stream with data, and sample rate is 16000
+streamObj.inputFinished();                   //tell engine you done with all data 
+while (rcgOjb.IsReady(streamObj)) {          //engine is ready
+
+                OnlineStream ssObj[]=new OnlineStream[1];
```

Comment:
Could you place this line outside of the while() loop so that you only need to create it once?